### PR TITLE
`Development`: Reduce transferred data when creating ratings

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
Creating and updating ratings resulted in a large response from the server. The whole path `result -> participation -> exercise -> course` was included, but this information was not needed on the client.

### Description
I changed the endpoint to only return the star rating integer.

For instructors, information about the participation and exercise are needed on the client, so I only removed the submission und course here.

### Steps for Testing
- As a student, create and update a rating. Make sure everything works
- As an instructor, make sure that the list of ratings works and shows all needed information (assessor, exercise, etc)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/dto/RatingDTO.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package de.tum.in.www1.artemis.web.rest.dto;
+
+import de.tum.in.www1.artemis.domain.Rating;
+
+public record RatingDTO(Integer rating, DomainObjectIdDTO result) {
+
+    public RatingDTO(Rating rating) {
```
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/dto/RatingDTO.java:7]: Why is the rating an integer?

Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/web/rest/dto/RatingDTO.java:7]: `Rating` is class that contains an integer `rating`, a start-rating from 1 to 5.

Comment (tobias-lippert) [reply src/main/java/de/tum/in/www1/artemis/web/rest/dto/RatingDTO.java:7]: And is it null if no rating is given?

Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/web/rest/dto/RatingDTO.java:7]: I used `Integer` here simply to mirror the same type as used in the `Rating` class, I did not look into it in that detail. I'd expect that there is just no rating object in that case.

**Review by tobias-lippert — DISMISSED** [review]
Code

**Review by JohannesStoehr — COMMENTED** [review]
Why didn't you adapt the endpoint on the client side to only use the DTO?

**Comment by Strohgelaender** [comment]
> Why didn't you adapt the endpoint on the client side to only use the DTO?

The JSON for the RatingDTO is identical to the Rating, just with minimal data. The client expects and still gets a rating, just the way how this object on the server side gets constructed is changed.

However, the client currently only needs the star value (int 1-5). If you want I can change the endpoint to only return that Integer.

**Review by MaximilianAnzinger — APPROVED** [review]
Code LGTM

File: src/main/webapp/app/exercises/shared/rating/rating.service.ts [hunk 2]
```diff
@@ -13,26 +13,28 @@ export class RatingService {
...
-    createRating(rating: Rating): Observable<Rating> {
-        return this.http.post<Rating>(this.ratingResourceUrl + `${rating.result!.id!}/rating/${rating.rating}`, null);
+    createRating(rating: number, resultId: number): Observable<number> {
+        return this.http.post<number>(this.ratingResourceUrl + `${resultId}/rating/${rating}`, null);
     }
 
     /**
      * Get rating for "resultId" Result
-     * @param ratingId - Id of Result who's rating is received
+     * @param resultId - ud of result who's rating is received
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/rating/rating.service.ts:25]:
```suggestion
     * @param resultId - id of result who's rating is received
```

**Review by JohannesStoehr — APPROVED** [review]
Thanks for changing the client endpoint as well. Code looks good to me

**Review by Hialus — APPROVED** [review]
Code looks good to me.
Just one small thing in your PR description. It says:
> I added a new DTO containing minimal information.

But this DTO no longer exists, so you should probably update the description to reflect this.

**Review by florian-glombik — APPROVED** [review]
Tested on ts1 for a text exercise; creating and updating a ranking worked fine from a student perspective, and the page was loaded properly!

The rating was displayed properly in the tutor leaderboard for the exercise.

**Review by Kroko-fant — APPROVED** [review]
tested on ts5

**Review by MaximilianAnzinger — APPROVED** [review]
Reapprove Code, also manually tested during testing session

**Review by laurenzfb — APPROVED** [review]
Tested on TS5, worked nicely

**Pull request merged**